language-design.md

```diff
@@ -49,6 +68,11 @@ In what way is an API a language?
+
+Furthermore, unlike a general purpose programming language, an API has a very small focus. Convoluting an API with general functionality is almost always a detriment. It is for this reason, Joshua’s advice often follows the KISS paradigm (Keep It Simple Stupid). While Joshua’s advice of “Keep APIs free of implementation details,” “When in doubt, leave it out,” and “Minimize accessibility; when in doubt, make it private” [Bloch, 2006, 506], is all given in the context of good API design, we can also validly apply these recommendations to a focused DSL. For instance, all of these points can also be made in reference to the design philosophy of ContextFree. Specifically, although ContextFree is a powerful tool for designing digital art, it is limited to just this task. That is, ContextFree intentionally does not enable users to access low level features and implementation details or complete non-digital art tasks.
+
+Finally, the purposes for creating an API or DSL shares many overlaps. Most notably although a DSL cannot enable a user to do something the host or implementation language cannot do, it can make it a lot easier for users. For instance, anything a user can do with a Bash Script can be done in raw C. That being said, for the specific domain of file and process manipulation, a Bash Script will often enable these tasks to be completed much easier and faster. Similarly, since an API enables code reusability and abstraction, an API can enable a developer to have a faster and easier development process since they do not have to fully re-implement the API’s content. Because of this, however, APIs need to be well designed so they actually simply the development process. In the advice of Bloch, “APIs should be self-documenting” and “Strive for intelligibility, consistency, and symmetry” [Bloch, 2006, 506]. It is for these reasons, like a good DSL, an API should be mostly intuitive making the targeted domain task simpler.
```

Comment:
Again, a good example. I don't think I have much to say here - I agree with all of your reasons for their similarity. We gave similar reasons, and I'm sure there are plenty of other examples that could be used to further bolster the apparent similarity of DSLs and APIs, but I agree with everything you've written here.